opencsp/app/sofast/SofastServer.py

```diff
@@ -0,0 +1,224 @@
+                        measurement = state.last_measurement_fringe
+                        state.system_fringe
+                if measurement is not None:
+                    ret.update(
+                        {
+                            "focal_length_x": measurement.focal_length_x,
+                            "focal_length_y": measurement.focal_length_y,
+                            "slope_error_x": np.average(measurement.slopes_error[0]),
+                            "slope_error_y": np.average(measurement.slopes_error[1]),
+                            "slope_error": np.average(measurement.slopes_error),
+                            "slope_stddev": np.std(measurement.slopes_error),
+                        }
+                    )
+                else:
+                    ret["error"] = "Fringe measurement is not ready"
+                    ret["trace"] = "SofastServer.get_response::get_results_fringes"
+                    response_code = 409
+
+            else:
+                ret["error"] = f"Unknown action \"{action}\""
+                ret["trace"] = "SofastServer.get_response::N/A"
+                response_code = 404
+
+        except Exception as ex:
+            lt.error("Error in SofastServer with action " + action + ": " + repr(ex))
+            ret["error"] = (repr(ex),)
+            ret["trace"] = "".join(format_exception(ex))
+            response_code = 500
+
+        # sanity check: did we synchronize the error and response_code?
+        if response_code != 200:
+            if ret["error"] is None:
+                lt.error_and_raise(
+                    RuntimeError,
+                    f"Programmer error in SofastServer.get_response({action}): "
+                    + f"did not correctly set 'error' to match {response_code=}!",
+                )
+        if ret["error"] is not None:
+            if response_code == 200:
+                lt.error_and_raise(
+                    RuntimeError,
+                    f"Programmer error in SofastServer.get_response({action}): "
+                    + f"did not correctly set response_code to match {ret['error']=}!",
+                )
+
+        return response_code, json.dumps(ret)
+
+
+if __name__ == "__main__":
+    port = 8000
+
+    # Set up the logger
+    log_output_dir = opencsp_settings["sofast_server"]["log_output_dir"]
+    if log_output_dir is not None and ft.directory_exists(log_output_dir):
+        log_name_ext = "SofastServer_" + tdt.current_date_time_string_forfile() + ".log"
+        log_path_name_ext = os.path.join(log_output_dir, log_name_ext)
+        lt.logger(log_path_name_ext)
+
+    # Start the server
+    lt.warn(
+        "Warning in SofastServer: this server is unsecured. "
+        + f"It is suggested that you restrict outside access to port {port} of the host computer."
+    )
+    lt.info(f"Starting server on port {port}...")
+    server = HTTPServer(("0.0.0.0", port), SofastServer)
```

Comment:
Consider storing "0.0.0.0" as a local variable above near where `port` is defined.

Reply:
Yep, this is a good idea and one I will implement.

Hey, sorry about the confusion. This PR isn't ready for review yet (if I assigned you already that was a mistake). I'll let you know when it is ready.